platform/javascript/export/export.cpp

```diff
@@ -536,23 +561,23 @@ Error EditorExportPlatformJavaScript::run(const Ref<EditorExportPreset> &p_prese
 		return OK;
 	}
 
-	String basepath = EditorSettings::get_singleton()->get_cache_dir().plus_file("tmp_js_export");
-	String path = basepath + ".html";
-	Error err = export_project(p_preset, true, path, p_debug_flags);
+	const String basepath = EditorSettings::get_singleton()->get_cache_dir();
+	Error err = export_project(p_preset, true, basepath.plus_file("tmp_js_export.html"), p_debug_flags);
 	if (err != OK) {
 		// Export generates several files, clean them up on failure.
-		DirAccess::remove_file_or_error(basepath + ".html");
-		DirAccess::remove_file_or_error(basepath + ".js");
-		DirAccess::remove_file_or_error(basepath + ".pck");
-		DirAccess::remove_file_or_error(basepath + ".png");
-		DirAccess::remove_file_or_error(basepath + ".wasm");
+		DirAccess::remove_file_or_error(basepath.plus_file("tmp_js_export.html"));
+		DirAccess::remove_file_or_error(basepath.plus_file("tmp_js_export.js"));
+		DirAccess::remove_file_or_error(basepath.plus_file("tmp_js_export.pck"));
+		DirAccess::remove_file_or_error(basepath.plus_file("tmp_js_export.png"));
+		DirAccess::remove_file_or_error(basepath.plus_file("tmp_js_export.wasm"));
```

Comment:
Any specific reason for the above changes? I'd prefer to keep a `String` with the name and just add extensions if possible.

Reply:
The favicon doesn't share the same basename as the other files, so I thought we could do it this way instead. Otherwise, I'd have to define another variable that contains the cache directory (not a huge issue, but it doesn't help with readability).

Reply:
I would prefer not to have a the export name copy-pasted 4 times (which is also inconsistent with what we do in `EditorHTTPServer`).
You don't really need an extra variable, you could call `EditorSettings::get_singleton()->get_cache_dir()` again, but yeah, also having a `const String cache_dir` variable is fine. Your choice.